src/cleanfigure.m

```diff
@@ -484,9 +485,12 @@ function simplifyLine(meta, handle, targetResolution)
                 mask      = opheimSimplify(x, y, tol);
                 % Remove all those with mask==0 respecting the number of
                 % data points in the previous segments
-                id_remove = find(mask==0) + lineStart(ii) - 1;
+                id_remove{ii} = find(mask==0) + lineStart(ii) - 1;
```

Comment:
At this point, you do not need the `find()` and the `lineStart` shifts, and you can save the index directly. The removeData function should work with the logical index too.